tools/release_generator/release.py

```diff
@@ -225,18 +226,29 @@ def makeSubDirectoriesFromList(directoryList, upperDirectory, HigherLevel):
         makeReleaseNoteSubfolders(combinedDirectory, HigherLevel)
 
 
+def subfolder_creation(directory, HigherLevel, folder):
+    subfolderName = HigherLevel / directory / folder
+    subfolderName.mkdir(parents=True, exist_ok=True)
+    makeGitkeep(subfolderName)
+
+
 def makeReleaseNoteSubfolders(directory, HigherLevel):
     directoryStr = str(directory)
     for folder in subfolders:
         if "Muon" in directoryStr:
             for single_folder in muon_subfolders:
-                subfolderName = HigherLevel / directory / single_folder
-                subfolderName.mkdir(parents=True, exist_ok=True)
-                makeGitkeep(subfolderName)
+                subfolder_creation(directory, HigherLevel, single_folder)
+        if "Framework" in directoryStr:
+            if "Algorithm" in directoryStr:
+                for single_folder in deprecated_subfolders:
+                    subfolder_creation(directory, HigherLevel, single_folder)
+            if "Fit_Functions" in directoryStr:
```

Comment:
I think this would better with an `or` instead of the two `if` statements, that would remove the code duplication. I mean e.g. 
``` python
if "Algorithm" in directoryStr or "Fit_Functions" in directoryStr:
```